src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -0,0 +1,205 @@
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+
+            checkBulkScanReferenceIsSet(
+                (Map<String, ?>) updateResponse.caseDetails.caseData,
+                exceptionRecord.id
+            );
+
+            updateCaseInCcd(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord,
+                updateResponse.caseDetails,
+                existingCase
+            );
+
+            log.info(
+                "Successfully updated case for {} with case ID {} from exception record {}",
+                configItem.getService(),
+                existingCase.getId(),
+                exceptionRecord.id
+            );
+        } catch (InvalidCaseDataException exception) {
+            throw new CallbackException("Failed to update case", exception);
+        } catch (PaymentsPublishingException exception) {
+            log.error(
+                "Failed to send update to payment processor for {} exception record {}",
+                configItem.getService(),
+                exceptionRecord.id,
+                exception
+            );
+
+            throw new CallbackException("Payment references cannot be processed. Please try again later", exception);
+        } catch (Exception exception) {
+            // log happens individually to cover update/ccd cases
+            throw new CallbackException("Failed to update case", exception);
+        }
+    }
+
+    @SuppressWarnings("squid:S2139") // exception handle + logging
+    private long updateCaseInCcd(
+        String idamToken,
+        String s2sToken,
+        String userId,
+        ExceptionRecord exceptionRecord,
+        CaseUpdateDetails caseUpdateDetails,
+        CaseDetails existingCase
+    ) {
+        try {
+            StartEventResponse eventResponse = coreCaseDataApi.startForCaseworker(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getCaseTypeId(),
+                // when onboarding remind services to not configure about to submit callback for this event
+                caseUpdateDetails.eventId
+            );
+
+            return coreCaseDataApi.submitForCaseworker(
+                idamToken,
+                s2sToken,
+                userId,
+                exceptionRecord.poBoxJurisdiction,
+                existingCase.getCaseTypeId(),
+                true,
+                CaseDataContent
+                    .builder()
+                    .caseReference(exceptionRecord.id)
+                    .data(caseUpdateDetails.caseData)
+                    .event(Event
+                        .builder()
+                        .id(eventResponse.getEventId())
+                        .summary("Case updated")
```

Comment:
Worth making it something like `Case updated based on exception record` or sth like this. Ideally, to be provided by Ade.